null_safety_examples/_animation/radial_hero_animation/main.dart

```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that
+// can be found in the LICENSE file.
+
+// A "radial transformation" as defined here:
+// https://material.io/guidelines/motion/transforming-material.html#transforming-material-radial-transformation
+
+// In this example, the destination route (which completely obscures
+// the source route once the page transition has finished),
+// displays the Hero image in a Card containing a column of two
+// widgets: the image, and some descriptive text.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart' show timeDilation;
+
+class Photo extends StatelessWidget {
+  Photo({ Key? key, required this.photo, this.onTap }) : super(key: key);
+
+  final String photo;
+  final VoidCallback? onTap;
+
+  Widget build(BuildContext context) {
+    return Material(
+      // Slightly opaque color appears where the image has transparency.
+      color: Theme.of(context).primaryColor.withOpacity(0.25),
+      child: InkWell(
+        onTap: onTap,
+        child: LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints size) {
+            return Image.asset(
+              photo,
+              fit: BoxFit.contain,
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class RadialExpansion extends StatelessWidget {
+  RadialExpansion({
+    Key? key,
+    required this.maxRadius,
+    this.child,
+  }) : clipRectSize = 2.0 * (maxRadius / math.sqrt2),
+       super(key: key);
+
+  final double maxRadius;
+  final clipRectSize;
+  final Widget? child;
+
+  @override
+  Widget build(BuildContext context) {
+    return ClipOval(
+      child: Center(
+        child: SizedBox(
+          width: clipRectSize,
+          height: clipRectSize,
+          child: ClipRect(
+            child: child,
```

Comment:
Does ClipRect really allow an optional child?

Reply:
It is indeed, although I agree it probably shouldn't :)